js/blocks/DictionaryBlocks.js

```diff
@@ -0,0 +1,236 @@
+                    === undefined
+            ) {
+                return JSON.stringify(dictionary);
+            } else if (
+                logo.blocks.blockList[logo.blocks.blockList[blk].connections[0]]
+                    .name.includes("transform")
+            ) {
+                // only return dictionary if connected to transformer
+                return dictionary;
+            } else {
+                return JSON.stringify(dictionary);
+            }
+        }
+    }
+
+    class DictionaryPairBlock extends FlowBlock {
+        constructor() {
+            super("dictPair", _("dictionary pair"));
+            this.setPalette("dictionary");
+            this.beginnerBlock(true);
+
+            this.setHelpString([
+                _("The dictionary pair block pairs a keyword with a value in a dictionary."),
+                "documentation",
+                ""
+            ]);
+
+            this.formBlock({
+                args: 2,
+                argLabels: [
+                    _("keyword"),
+                    _("value")
+                ],
+                argTypes: ["textin", "anyin"],
+                defaults: ["number", 4]
+            })
+        }
+
+        flow(args, logo, turtle, blk, receivedArg) {
+            var blks = logo.blocks.blockList;
+            var key = blks[blks[blk].connections[1]].value;
+            var value = blks[blks[blk].connections[2]].value;
+            logo.textMsg(`${key}: ${value}`);
+        }
+    }
+
+    class NamedDictionaryBlock extends LeftBlock {
+        constructor() {
+            super("namedDictionary", _("dictionary"));
+            this.setPalette("dictionary");
+            this.beginnerBlock(true);
+
+            this.setHelpString([
+                _("The dictionary block returns a dictionary."),
+                "documentation",
+                ""
+            ]);
+
+            this.extraWidth = 20;
+            this.formBlock({
+                args: 1,
+                argTypes: ['textin'],
+                argLabels: ['name'],
+                defaults: [_('dictionary')],
+                flows: {
+                    labels: [""],
+                    type: "flow"
+                },
+            });
+        }
+
+        arg(logo, turtle, blk) {
+            var dictionary = {};
+            var blks = logo.blocks.blockList;
+            var currBlock = blks[blk];
+
+            while ((currBlock.connections.slice(-1)[0]) !== null) {
+                var nextBlock = blks[currBlock.connections.slice(-1)[0]];
+                if (nextBlock.name === "dictPair") {
+                    var key = blks[nextBlock.connections[1]].value;
+                    var value = blks[nextBlock.connections[2]].value;
+                    // temporary fix
+                    if (value === "true") {
+                        value = true;
+                    } else if (value === "false") {
+                        value = false;
+                    }
```

Comment:
can use `value = (value === "true")` here

Reply:
I'd like boolean values to go through as booleans, but the boolean blocks can't be used as arguments yet, so I made it so that the string values `"true"` and `"false"` will be converted to booleans. Everything else should be passed through normally though – `"cat"` should still be `"cat"`.